crates/egui_kittest/src/renderer.rs

```diff
@@ -0,0 +1,81 @@
+use egui::{Context, FullOutput, TexturesDelta};
+use image::RgbaImage;
+
+pub trait TestRenderer {
+    /// We use this to pass the glow / wgpu render state to [`eframe::Frame`].
+    #[cfg(feature = "eframe")]
+    fn setup_eframe(&self, _cc: &mut eframe::CreationContext<'_>, _frame: &mut eframe::Frame) {}
+
+    /// Handle a [`TexturesDelta`] by updating the renderer's textures.
+    fn handle_delta(&mut self, delta: &TexturesDelta);
+
+    /// Render the [`crate::Harness`] and return the resulting image.
+    ///
+    /// # Errors
+    /// Returns an error if the rendering fails.
+    fn render(&mut self, ctx: &Context, output: &FullOutput) -> Result<RgbaImage, String>;
+}
+
+/// A lazy renderer that initializes the renderer on the first render call.
+///
+/// By default, this will create a wgpu renderer if the wgpu feature is enabled.
+pub enum LazyRenderer {
+    Uninitialized {
+        texture_ops: Vec<egui::TexturesDelta>,
+        builder: Option<Box<dyn FnOnce() -> Box<dyn TestRenderer>>>,
+    },
+    Initialized {
+        renderer: Box<dyn TestRenderer>,
+    },
+}
+
+impl Default for LazyRenderer {
+    fn default() -> Self {
+        #[cfg(feature = "wgpu")]
+        return Self::new(crate::wgpu::WgpuTestRenderer::new);
+        #[cfg(not(feature = "wgpu"))]
+        return Self::Uninitialized {
+            texture_ops: Vec::new(),
+            builder: None,
+        };
+    }
+}
```

Comment:
Does this default really make sense? I would have expected the default to be always uninitialized because after all it's supposed to be lazy

Reply:
misread, it _is_ uninitialized